mod-source-record-storage-server/src/main/java/org/folio/services/util/AdditionalFieldsUtil.java

```diff
@@ -509,4 +570,22 @@ private static boolean isNotProtected(List<MarcFieldProtectionSetting> fieldProt
       .filter(setting -> setting.getField().equals(ANY_STRING) || setting.getField().equals(field.getTag()))
       .noneMatch(setting -> setting.getData().equals(ANY_STRING) || setting.getData().equals(field.getData()));
   }
+
+  private static org.marc4j.marc.Record computeMarcRecord(Record record) {
+    if (record != null
+      && record.getParsedRecord() != null
+      && !org.apache.commons.lang3.StringUtils.isBlank(record.getParsedRecord().getContent().toString())) {
+      try {
+        return parsedRecordContentCache.get(record.getParsedRecord().getContent());
+      } catch (Exception e) {
+        LOGGER.error("Error during the transformation to marc record", e);
+        MarcReader reader = buildMarcReader(record);
+        if (reader.hasNext()) {
+          return reader.next();
+        }
```

Comment:
Hello @KyryloShulzhenko, do we need here to try to parse a record one more time in case of exception? Because it looks like this "catch" block can be called due to record parsing error in the `parsedRecordContentCacheLoader`
block, while calling parsedRecordContentCache.get() method of the cache.

Reply:
Hello @RuslanLavrov  parsedRecordContentCache.get() in some cases don`t parse record and throw exception. This code help to parse record

Reply:
in such case, propose to set the "warn" logging level here at the "catch" block, since the method can return a successfully parsed record